### PR TITLE
mingw-w64-osgearth: Initial commit. v2.7

### PR DESCRIPTION
**Pull request by parched**
This adds [osgearth](http://osgearth.org)

### CODE REVIEW

File: mingw-w64-osgearth/0003-Build-fix-against-OSG-3.4.0-and-OSG-3.5.0.patch [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
++#endif
+ 
+         protected:
+ 
+@@ -51,7 +55,7 @@ namespace osgEarth
+ 
+     /**
+      * ElevationQuery (EQ) lets you query the elevation at any point on a map.
+-     * 
++     *
```
Comment (mingwandroid) [inline mingw-w64-osgearth/0003-Build-fix-against-OSG-3.4.0-and-OSG-3.5.0.patch]: Hi @parched, there's quite a few of these redundant lines in this patch, would you mind sorting them out, if for no other reason than to satiate my obsessive compulsive tenancies?

File: mingw-w64-osgearth/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,87 @@
+# Maintainer: James Duley <jagduley gmail>
+
+_realname=osgearth
```
Comment (mingwandroid) [inline mingw-w64-osgearth/PKGBUILD]: please add pkgbase after this line.

File: mingw-w64-osgearth/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  patch -p1 -i ${srcdir}/0001-Fix-mingw-debug-build.patch
+  patch -p1 -i ${srcdir}/0002-Forced-include-headers-for-Qt5-moc-on-files-without-.patch
+  patch -p1 -i ${srcdir}/0003-Build-fix-against-OSG-3.4.0-and-OSG-3.5.0.patch
+}
+
+build() {
+  plain "No build step"
+}
+
+package_release() {
```
Comment (mingwandroid) [inline mingw-w64-osgearth/PKGBUILD]: We don't normally go down the route of debug and release packages, it's too much package immanence work for Alexey.

options=('!strip' 'debug') and if check_options "debug" "y"; is the usual mechanism.

Can you go down this route? There's a good amount of reference. You can use agm2 to search quickly in the PKGBUILDs.

File: mingw-w64-osgearth/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+package_debug() {
+  depends=("${MINGW_PACKAGE_PREFIX}-${_realname}-svn=$pkgver" "${MINGW_PACKAGE_PREFIX}-OpenSceneGraph-debug")
+  provides=("${MINGW_PACKAGE_PREFIX}-${_realname}-debug")
+  conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}-debug")
+
+  [[ -d ${srcdir}/build-debug ]] && rm -rf ${srcdir}/build-debug
+  mkdir -p ${srcdir}/build-debug && cd ${srcdir}/build-debug
+
+  ${MINGW_PREFIX}/bin/cmake.exe \
```
Comment (mingwandroid) [inline mingw-w64-osgearth/PKGBUILD]: cmake packages work more idiomatically if you configure as:
```
MSYS2_ARG_CONV_EXCL=-DCMAKE_INSTALL_PREFIX= \
    ${MINGW_PREFIX}/bin/cmake.exe \
    -G"MSYS Makefiles" \
    -DCMAKE_INSTALL_PREFIX:PATH=${MINGW_PREFIX}
```

.. then install as:
```
make install DESTDIR="${pkgdir}"
```

File: mingw-w64-osgearth/0003-Build-fix-against-OSG-3.4.0-and-OSG-3.5.0.patch (hunk 1 above)
Comment (parched) [inline mingw-w64-osgearth/0003-Build-fix-against-OSG-3.4.0-and-OSG-3.5.0.patch]: Ok I fix this. As it stands, it is just two squashed commits from upstream.

File: mingw-w64-osgearth/PKGBUILD (hunk 3 above)
Comment (parched) [inline mingw-w64-osgearth/PKGBUILD]: Ok thanks, I will change it. I just copied this from the osgocean package I think. I will look at other cmake packages and see how they do it.

**Pull request merged**